tests/locale/keymap_or_locale.pm

```diff
@@ -76,18 +71,30 @@ sub verify_default_keymap_x11 {
 }
 
 sub run {
-    # uncomment in case of different keyboard than us is used during installation ( feature not ready yet )
-    # my $expected   = get_var('INSTALL_KEYBOARD_LAYOUT','us');
-    my $expected       = 'us';
-    my %keystroke_list = (
-        us => '`1234567890-=~!@#$%^&*()_+',
-        fr => '²&é"(-è_çà)=~1234567890°+',
-        de => '1234567890ß°!"§$%&/()=?',
-        cz => ';+ěščřžýáíé=1234567890%'
-    );
-    my $keystrokes = $keystroke_list{$expected};
-
-    if (check_var('DESKTOP', 'textmode')) {
+    my $expected = get_var('INSTALL_KEYBOARD_LAYOUT', 'us');
+    # feature of switching keyboard during installation is not ready yet,
+    # so when typing, expected result is as follows:
+    # fr => '²&é"(-è_çà)=~1234567890°+',
+    # de => '1234567890ß°!"§$%&/()=?',
+    # cz => ";+ěščřžýáíé=';1234567890%"
+    my $keystrokes = '`1234567890-=!@6~!@#$%^&*()_+';
+
+    if (get_var('INSTALL_KEYBOARD_LAYOUT')) {
```

Comment:
I guess you want to check if layout is not `us` here, because such condition is definitely misleading. As your current AY test does exactly text mode installation, so I'm a bit lost here =) I guess we just could have adjusted `verify_default_keymap_textmode` and potentially rename it, as it's not `default`.

Reply:
How it was before it might be intended to do that, but reality is that  those string in the hash just represent key presses in the 2th row from the top of the keyboard (one without SHIFT and another one with it) but not the equivalent, for example in `us` first we type the numbers and in `cz` we would need to type the second pass first with `!` to get the equivalent to `1` and have the same needle. Also the string itself is not portable, `=` symbol in `cz` keyboard is not a symbol, it is a helper to put other symbol in top of the letter like `ě`. In general I didn't find worth to keep those string as they are not used and are confusing for several reason.
Regarding `verify_default_keymap_textmode` yes, you're right, I will adjust the function and fix the mistake with the non-us condition.

Reply:
Sounds good to me, let's make it this way then ;)